### PR TITLE
Implement Controller Specs

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Description
- Install Controllers tester
- Write Tests
- Fix Linter errors

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'works! (now write some real specs)' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('posts/show')
+      expect(response.body).to include('Post 1 from User 1')
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/requests/posts_spec.rb]: - [ ] Great job writing tests for your controllers. However, at this moment, the tests are failing. Kindly look into them and fix them.

### Posts errors
post-error

post-error-1

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  describe 'GET /users/1' do
+    it 'works! (now write some real specs)' do
+      get user_path(1)
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('users/show')
+      expect(response.body).to include('User 1')
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/requests/users_spec.rb:21]: - [ ] Same thing here

### Users errors
user-errors

user-error-1

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @IsmailChangezi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kamba56 — APPROVED** [review]
Hi @IsmailChangezi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] great job using rspec to test controllers👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**